openassessment/xblock/static/js/spec/lms/oa_staff_area.js

```diff
@@ -281,6 +247,101 @@ describe('OpenAssessment.StaffAreaView', function() {
+                staffArea.handleCommentChanged();
+                expect(staffArea.cancelSubmissionEnabled()).toBe(false);
+
+                // Response is not blank --> cancel submission button enabled
+                staffArea.comment('Cancellation reason.');
+                staffArea.handleCommentChanged();
+                expect(staffArea.cancelSubmissionEnabled()).toBe(true);
+            });
+
+            it('submits the cancel submission comments to the server', function() {
+                // Show the staff area for the test student
+                var staffArea = createStaffArea();
+                chooseStudent(staffArea, 'testStudent');
+
+                // Cancel the student's submission
+                staffArea.comment('Cancellation reason.');
+                server.studentTemplate = 'oa_staff_cancelled_submission.html';
+                staffArea.cancelSubmission('Bob');
+
+                // Verify that the student view reflects the cancellation
+                expect($($('.staff-info__student__response p', staffArea.element)[0]).text().trim()).toBe(
+                    'Learner submission removed by staff on October 1, 2015 04:53 UTC'
+                );
+                expect($($('.staff-info__student__response p', staffArea.element)[1]).text().trim()).toBe(
+                    'Comments: Cancelled!'
+                );
+            });
+        });
+
+        describe('Staff Grade Override', function() {
+            var fillAssessment = function($assessment) {
+                $('#staff__assessment__rubric__question--2__feedback', $assessment).val('Text response');
+                $('.question__answers', $assessment).each(function(element) {
+                    $('input[type="radio"]', this).first().click();
+                });
+            };
+
+            it('enables the submit button when all required fields are specified', function() {
+                var staffArea = createStaffArea(),
+                    $assessment, $submitButton;
+                chooseStudent(staffArea, 'testStudent');
+                $assessment = $('.wrapper--staff-assessment', staffArea.element)
+                $submitButton = $('.action--submit', $assessment);
+                expect($submitButton).toHaveClass('is--disabled');
+                fillAssessment($assessment);
+                expect($submitButton).not.toHaveClass('is--disabled');
+            });
+
+            it('can submit a staff grade override', function() {
+                var staffArea = createStaffArea(),
+                    $assessment, $submitButton;
+                chooseStudent(staffArea, 'testStudent');
+                $assessment = $('.wrapper--staff-assessment', staffArea.element)
+                $submitButton = $('.action--submit', $assessment);
+                fillAssessment($assessment);
+
+                // Submit the assessment
+                server.studentTemplate = 'oa_staff_cancelled_submission.html';
```

Comment:
This test is testing the wrong thing-- cancellation instead of staff override (see expects below as well!).